Civi/Api4/Generic/Traits/DAOActionTrait.php

```diff
@@ -398,6 +398,7 @@ protected function updateWeight(array &$record) {
         $oldWeight = civicrm_api4($this->getEntityName(), 'get', [
           'select' => [$weightField],
           'where' => $where,
+          'checkPermissions' => FALSE,
```

Comment:
@eileenmcnaughton whilst I think FALSE is probably ok maybe better to use $this->getCheckPermissions(); like L320

Reply:
@seamuslee001 OK - what are your thoughts against putting it against the rc - it's not really a regression but I'm trying to update tests from apiv3 to apiv4 & hitting this & it gets kinda complicated publishing an extension with tests that won't work until x is released - at least I could bring that back by a month...

Reply:
@seamuslee001 I agree with all that - but it sounds like 'OK at a pinch' - so I changed base